File: iterator_pattern/DtuMember.php
<?php
require_once('./Collection.php');
require_once('./DtuIterator.php');
require_once('./Person.php');

class DtuMember implements Collection {
    private $members = array();

    public function __construct(array $members = null) {
        if(!is_null($members)) {
            $this->members = $members;
        }
    }

    // イテレータを返す
    public function getIterator() {
        return new DtuIterator($this->members);
    }

    // DTUに加入
    public function joinDtu(Person $person) {
        $this->members[] = $person;
    }
}

File: iterator_pattern/java/DtuMember.java
import java.util.ArrayList;
import java.util.List;

class DtuMember implements Collection {
  private List<Person> members;

  public DtuMember(Person[] members) {
    this.members = new ArrayList<Person>();
    // 引数で受け取った配列をプロパティに格納
    for(Person member : members) {
      this.members.add(member);
    }
  }

  public DtuIterator getIterator() {
    return new DtuIterator(this.members);
  }

  public void joinDtu(Person member) {
    this.members.add(member);
  }
}
